modules/core/src/persistence_xml.cpp

```diff
@@ -309,7 +309,7 @@ class XMLEmitter : public FileStorageEmitter
         if( !multiline )
         {
             ptr = fs->resizeWriteBuffer( ptr, len + 9 );
-            sprintf( ptr, "<!-- %s -->", comment );
+            snprintf( ptr, len + 9, "<!-- %s -->", comment );
```

Comment:
Looks like it should be `+ 10` (+ '\0` char).

Including buffer allocation above (it is a bug).

Reply:
Good catch!  Patch updated...